Creating-Rules.md

```diff
@@ -1,51 +1,74 @@
+The return value from a rule is always a boolean that describes whether `answer` and `inputs` obey the rule. For example, the `IsEquivalentTo` rule returns whether the user's inputs are equivalent to the creator-configured answer.
 
-This function has access to `answer`, as well as any parameters defined in the rule description string. In the previous example, the parameters `a` and `b` can be accessed using `inputs.a` and `inputs.b`. Here is an example of what the corresponding evaluation function might look like:
+For example, suppose we have a rule `IsBetweenInclusive` with the description `is at least {{x|Real}} and at most {{y|Real}}`. Then our evaluation function could be:
 
-```
-    ruleName: function(answer, inputs) {
-        return inputs.a <= answer <= inputs.b
-    }
+```ts
+export class MyInteractionRulesService {
+  IsBetweenInclusive(
+      answer: MyInteractionAnswer,
+      inputs: MyInteractionInputs): boolean {
+    return inputs.x <= answer <= inputs.y
+  }
+}
 ```
 
-The evaluation function may not use any other external parameters (such as the reader's current card, and so on) to compute its result; everything must be passed in through the subject or be defined when the rule is first initialized.
+Note that the evaluation function must not use any external parameters, for example the reader's current card. The `inputs` and `answer` must be sufficient to determine whether the rule is satisfied.
 
-## Adding Tests ##
+## Adding a New Rule
 
-You will need to create a `MyInteractionRulesServiceSpec.js` file containing tests for rules defined in the `myInteractionRulesService` factory service. Create a separate test for each rule and check the various corner cases to make sure that rules are defined correctly. You are strongly encouraged to do this for any new rule that you create, so that you can ensure that the rule works properly and returns the correct results on a variety of inputs.
+Suppose you have just [[created a new interaction|Creating-Interactions]], say `MyInteraction`, and now you want to define the rules associated with it. You should:
 
-## Contributing New Rules ##
+1. In `extensions/interactions/rule_templates.json`, add a dictionary under the key `MyInteraction`. You can skip this step if you are adding a rule to an existing interaction.
+2. For each rule you want to create add an appropriate key that names the rule. The corresponding value should be a dictionary with a single key, `description`, whose value is a string describing the rule. The string should contain parameters as substrings of the form `{{paramName|ParamType}}`. See the [descriptions section above](#rule-descriptions) for details.
```

Comment:
Unfortunately 'param'/'parameters' is a term of art that is used for something else in Oppia (see http://oppia.github.io/#/Parameters). Although it's being deprecated, references to params still exist in the codebase. 

I think we typically call these "rule inputs", would it be possible to use that terminology instead?

Reply:
done